node/src/components/block_synchronizer/tests.rs

```diff
@@ -95,35 +180,77 @@ fn check_sync_global_state_event(event: MockReactorEvent, block: &Block) {
+    block: &Block,
+    validator_keys_iter: I,
+) {
+    for secret_key in validator_keys_iter {
+        // Register a finality signature
+        let signature = FinalitySignature::create(
+            *block.hash(),
+            block.header().era_id(),
+            secret_key.as_ref(),
+            PublicKey::from(secret_key.as_ref()),
+        );
+        assert!(signature.is_verified().is_ok());
+        assert!(builder.register_finality_signature(signature, None).is_ok());
+    }
+}
+
+impl BlockSynchronizer {
+    // Create an initialized block synchronizer with default config and maximum 5 simultaneous peers
+    fn new_initialized(rng: &mut TestRng, validator_matrix: ValidatorMatrix) -> BlockSynchronizer {
+        let mut block_synchronizer = BlockSynchronizer::new(
+            Config::default(),
+            Arc::new(Chainspec::random(rng)),
+            MAX_SIMULTANEOUS_PEERS as u32,
+            validator_matrix,
+            &prometheus::Registry::new(),
+        )
+        .expect("Failed to create BlockSynchronizer");
+
+        <BlockSynchronizer as InitializedComponent<MainEvent>>::set_state(
+            &mut block_synchronizer,
+            ComponentState::Initialized,
+        );
+
+        block_synchronizer
+    }
+
+    fn forward_builder(&self) -> &BlockBuilder {
+        self.forward.as_ref().expect("Forward builder missing")
+    }
+}
+
 #[tokio::test]
 async fn global_state_sync_wont_stall_with_bad_peers() {
     let mut rng = TestRng::new();
     let mock_reactor = MockReactor::new();
-
-    // Set up a random block that we will use to test synchronization
-    let block = Block::random(&mut rng);
-
-    // Set up a validator matrix for the era in which our test block was created
-    let mut validator_matrix = ValidatorMatrix::new_with_validator(ALICE_SECRET_KEY.clone());
-    validator_matrix.register_validator_weights(
-        block.header().era_id(),
-        iter::once((ALICE_PUBLIC_KEY.clone(), 100.into())).collect(),
+    let test_env = TestEnv::random(&mut rng).with_block(
+        TestBlockBuilder::new()
+            .era(1)
+            .deploys([Deploy::random(&mut rng)].iter())
+            .build(&mut rng),
     );
-
-    // Create a block synchronizer with a maximum of 5 simultaneous peers
-    let metrics_registry = Registry::new();
-    let mut block_synchronizer = BlockSynchronizer::new(
-        Config::default(),
-        Arc::new(Chainspec::random(&mut rng)),
-        5,
-        validator_matrix,
-        &metrics_registry,
-    )
-    .unwrap();
-
-    // Generate more than 5 peers to see if the peer list changes after a global state sync error
-    let num_peers = rng.gen_range(10..20);
-    let peers: Vec<NodeId> = random_peers(&mut rng, num_peers).iter().cloned().collect();
+    let peers = test_env.peers();
+    let block = test_env.block();
+    let validator_matrix = test_env.gen_validator_matrix();
+    let validators_private_keys = test_env.validator_keys();
```

Comment:
Can we rename "private keys" to "secret keys" throughout this file please?

(I'm overly nitpicky about this because there is a tendency to name variables `pk` for either public keys or private keys.  If we at least have a convention of favouring "secret key" over "public key" then such an abbreviation is a little less dangerous.)